fhttp/http_client.go

```diff
@@ -380,6 +380,20 @@ func newHTTPRequest(o *HTTPOptions) (*http.Request, error) {
 	}
 	//nolint:noctx // we pass context later in Run()/Fetch()
 	req, err := http.NewRequest(method, o.URL, body)
+	if err == nil { //nolint:nestif
+		// Additional validation for the URL so we abort early on fatal errors even for the std client.
+		// fixes #784
+		if req.URL == nil || req.URL.Host == "" {
```

Comment:
Doesn't it seem more logical to do this sanity check even before newHttpRequest? I.e., somewhere in the flag handling?

Reply:
flag handling? this can be called (is called) by UI handler and rapi so no flag - this is to catch/fix #784

Reply:
'whatever input' handling then :p 

More my point is; can't we check earlier than this point?

Reply:
it's the first thing done in NewStdClient

https://github.com/fortio/fortio/blob/master/fhttp/http_client.go#L559C6-L561

so there isn't really anything "earlier" no

Reply:
trying and failing to get github to show code preview from links, by hand:

```go
func NewStdClient(o *HTTPOptions) (*Client, error) {
	o.Init(o.URL) // also normalizes NumConnections etc to be valid.
	req, err := newHTTPRequest(o)
	if req == nil {
		return nil, err
	}
```

Reply:
in theory it could be inside Init but that wouldn't cover the case the url is changed after init etc